apps/studio/src/features/editing-experience/utils/convert.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  schema,
+  type IsomerNativeComponentProps,
+} from '@opengovsg/isomer-components'
+import { type JSONContent } from '@tiptap/react'
+import Ajv from 'ajv'
+
+export const validateAsProse = (
+  content: JSONContent,
+): IsomerNativeComponentProps => {
+  const ajv = new Ajv({ strict: false })
+  const subSchema = {
+    ...schema.components.native.prose,
+    components: schema.components,
+  }
+  const validate = ajv.compile<IsomerNativeComponentProps>(subSchema)
```

Comment:
note to reviewers: this is as good as a cast. changing the type here will cause the return type of `validate` to be that type, even if it's wrong. 

we should probably shift to typebox sooner rather than later.